src/main/java/eu/darkbot/api/entities/Npc.java

```diff
@@ -0,0 +1,14 @@
+package eu.darkbot.api.entities;
+
+public interface Npc extends Ship {
+
+    /**
+     * @return id of the npc
+     */
+    int getNpcId();
+
+    /**
+     * @return true if {@link} has {@code InstaShield} effect
+     */
+    boolean hasInstaShield();
```

Comment:
in the API should just use `hasEffect`, the ish flag was left for legacy reasons